lib/task/import/csvAuthorityRecordRelationImportTask.class.php

```diff
@@ -169,45 +169,51 @@ private function import($filepath, $indexDuringImport = false, $updateMode = fal
 
   private function importRow($sourceActorId, $targetActorId, $relationTypeId)
   {
-    // Check to see if a relation of the same type already exists
-    if ($relationId = $this->getRelation($sourceActorId, $targetActorId, $relationTypeId))
+    if (!empty($this->import->status['updateMode']))
     {
-      // Ignore this data if not updating
-      if (empty($this->import->status['updateMode']))
-      {
-        return;
-      }
-
-      // Delete existing relationship, then add new, if in "delete and replace" update mode
       if ($this->import->status['updateMode'] == 'delete-and-replace')
       {
-        $relation = QubitRelation::getById($relationId);
-        $relation->delete();
+        // Handle delete-and-replace updating by deleting all existing relations between source
+        // and target then adding new relation
+        $relations = $this->getRelations($targetActorId, $sourceActorId);
+        $relationsAlternate = $this->getRelations($sourceActorId, $targetActorId);
+
+        foreach (array_unique(array_merge($relations, $relationsAlternate)) as $relationId)
+        {
+          $relation = QubitRelation::getById($relationId);
+          $relation->delete();
+        }
 
         $this->addRelation($sourceActorId, $targetActorId, $relationTypeId);
       }
-      else
+      else if ($relationId = $this->getRelationByType($sourceActorId, $targetActorId, $relationTypeId))
       {
-        // In "match and update" mode then update the relation
+        // Handle match-and-update when a relation already exists
         $this->updateRelation($relationId, $sourceActorId, $targetActorId, $relationTypeId);
```

Comment:
This won't change the existing data at all, will it?  The match requires an exact match on source, target, and relation type so the update will just update the database with the exact same three values I think.

Reply:
The call to `setRelationFields` in `updateRelation` looks at current column values and uses those to set relation properties.